Sources/Growth/URP/ReferralData.swift

```diff
@@ -34,3 +34,67 @@ struct ReferralData {
     self.offerPage = json["offer_page_url"].string
   }
 }
+
+public struct AdAttributionData {
```

Comment:
I prefer keeping it as it is since I have custom error cases there and logging for it also entire module uses serialization same way.

The serial object struct moved to another file. fix at https://github.com/brave/brave-ios/pull/8194/commits/7eaec4a76215daa48b4cf9caa98414bd73cdc504